### DISCUSSION TITLE
Rotary encoder input from DAQ

### DISCUSSION THREAD

**Discussion by andrada08** [bonsai-rx/bonsai#1253] [Q&A]
Hi, I'm trying to read a rotary encoder/ quadrature counter input from a NI DAQ (USB-6210) board into Bonsai. I've connected the encoder to the counter pins on the DAQ and there is a straightforward way to configure this in the DAQ software and in MATLAB, where you can set the input type to 'Position' and it works. Is there a way to do this in Bonsai? I can see digital/analog inputs but no option for a counter.

**Comment by jkbhagatio** [comment]
ping @bruno-f-cruz any chance you know how to read a counter from a NI-DAQ USB device in Bonsai? I'm also curious about this.

**Reply by jkbhagatio** [reply]
@andrada08 you may be able to use a digital input for this, but would require some extra bonsai code to get the same functionality you get in in NI-DAQ mx MATLAB package. If you want we could try looking at this together over a video call sometime? I haven't really worked with the NI-DAQ package in Bonsai but happy to try and help

**Reply by andrada08** [reply]
@jkbhagatio sure that'd be great!

**Reply by petersaj** [reply]
For future reference - @andrada08 and I solved this by using an Arduino instead, easy and cheap. Attached sample arduino code and bonsai workflow (added .txt extension so it would upload). Note that the rotary encoder has to plug into arduino pins that are capable of interrupts (https://www.arduino.cc/reference/en/language/functions/external-interrupts/attachinterrupt/).

Thanks to Aman Saleem and Adam Tong for the code!

[arduino_rot_encoder.ino.txt](https://github.com/bonsai-rx/bonsai/files/11000600/arduino_rot_encoder.ino.txt)
[simple_wheel.bonsai.txt](https://github.com/bonsai-rx/bonsai/files/11000601/simple_wheel.bonsai.txt)

**Comment by glopesdev** [comment]
@andrada08 You can decode a quadrature counter by reading the data from the A/B pins using `AnalogInput` and then running this C# extension: https://gist.github.com/glopesdev/3070890ad944c0726ad2c99bbd9cbe82
